### ISSUE TITLE
Go 1.13 errors

### ISSUE THREAD

**Issue by djangulo** [quii/learn-go-with-tests#277]
**TL:DR** Go 1.13 has new error handling, @quii would you like to migrate the codebase?

Go 1.13 introduced new utilities of error handling (more details in the [go blog](https://blog.golang.org/go1.13-errors) and the [ErrorValueFAQ](https://github.com/golang/go/wiki/ErrorValueFAQ)).

Taking from the examples in the go blog:

Comparing errors:

```go
// Similar to:
//   if err == ErrNotFound { … }
if errors.Is(err, ErrNotFound) {
    // something wasn't found
}
```

Asserting error types:

```go
// Similar to:
//   if e, ok := err.(*QueryError); ok { … }
var e *QueryError
if errors.As(err, &e) {
    // err is a *QueryError, and e is set to the error's value
}
```

With the utility that now errors can be wrapped, similar to what [github.com/pkg/errors](https://github.com/pkg/errors) enables you to do.

Both `errors.Is` and `errors.As` have unwrapping functionality, meaning that they will scan the entire error chain contained inside an error (via an unmentioned interface that implements the `Unwrap` method).

```go
if err != nil {
    // Return an error which unwraps to err.
    // %w is a new verb that wraps `err`
    return fmt.Errorf("decompress %v: %w", name, err)
}
```

@quii Is this something you'd like to [slowly] migrate the codebase to?
